cmd/clusterctl/client/repository/components.go

```diff
@@ -304,9 +315,71 @@ func fixTargetNamespace(objs []unstructured.Unstructured, targetNamespace string
 		if util.IsResourceNamespaced(o.GetKind()) {
 			o.SetNamespace(targetNamespace)
 		}
+
+		if o.GetKind() == mutatingWebhookConfigurationKind || o.GetKind() == validatingWebhookConfigurationKind || o.GetKind() == customResourceDefinitionKind {
+			var err error
+			o, err = fixWebHookNamespaceReferences(o, targetNamespace)
+			if err != nil {
+				return nil, err
+			}
+		}
+		objs[i] = o
+	}
+	return objs, nil
+}
+
+func fixWebHookNamespaceReferences(o unstructured.Unstructured, targetNamespace string) (unstructured.Unstructured, error) {
+	annotations := o.GetAnnotations()
+	secretNamespacedName, ok := annotations["cert-manager.io/inject-ca-from"]
+	if ok {
+		secretName := strings.Split(secretNamespacedName, "/")[1]
```

Comment:
Let's raise an error if split doesn't return 2 parts (the secret name is misconfigured)